### DISCUSSION TITLE
Error at startup: nginx: [emerg] invalid IPv6 address in resolver "[fe80::1%eth0]"

### DISCUSSION THREAD

**Discussion by jdolderer** [ZoeyVid/NPMplus#1186] [Q&A] [answered]
Hello,

when trying to launch the container I get the following error:

`nginx: [emerg] invalid IPv6 address in resolver "[fe80::1%eth0]"`

It can be avoided by changing the corresponding entry in `/etc/resolv.conf ` from 

`nameserver fe80::1%eth0` to 
`nameserver fe80::1`. 

However this will lead to docker resources not being able to resolve properly, e.g. reaching dockerhub or installing dependecies (certbot plugin in this case), so I am stuck at this point.

It worth mentioning that I am running a Pihole instance with unbound to resolve DNS. 

Thanks for any input! 🙏

**Comment by Zoey2936** [comment]
Nginx just reads what inside /etc/resolv.conf, maybe you can give npmplus diffrent dns servers (without %eth0)?

**Reply by Zoey2936** [reply, marked as answer]
https://docs.docker.com/reference/compose-file/services/#dns

**Reply by jdolderer** [reply]
That actually did the trick, thank you!

I was a bit confused, as it has been working without any issues for npm before.

**Reply by Zoey2936** [reply]
upstream npm? yes they do it using a shellscript, I do it using an openresty config patch, thats why it works diffrent
